core/src/main/java/jenkins/monitor/JavaVersionRecommendationAdminMonitor.java

```diff
@@ -78,8 +139,55 @@ public HttpResponse doAct(@QueryParameter String no) throws IOException {
             disable(true);
             return HttpResponses.forwardToPreviousPage();
         } else {
-            return new HttpRedirect("https://www.jenkins.io/redirect/upgrading-jenkins-java-version-8-to-11");
+            return new HttpRedirect("https://jenkins.io/redirect/java-support/");
         }
     }
 
+    @NonNull
+    private static LocalDate getEndOfLife() {
+        LocalDate endOfLife = SUPPORTED_JAVA_VERSIONS.get(Runtime.version().feature());
+        return endOfLife != null ? endOfLife : LocalDate.MAX;
+    }
+
+    @NonNull
+    private static Period getDeprecationPeriod() {
+        return Period.between(LocalDate.now(), getEndOfLife());
+    }
+
+    @NonNull
+    private static Severity getSeverity() {
+        return getDeprecationPeriod().getYears() < 1 ? Severity.DANGER : Severity.WARNING;
+    }
+
+    /**
+     * @return The current feature-release counter.
+     * @see Runtime#version()
+     */
+    @Restricted(DoNotUse.class)
+    public int getJavaVersion() {
+        return Runtime.version().feature();
+    }
+
+    /**
+     * @return The end of life date for the current Java version in the system default time zone.
+     */
+    @Restricted(DoNotUse.class)
+    public Date getEndOfLifeAsDate() {
+        return Date.from(getEndOfLife().atStartOfDay(ZoneId.systemDefault()).toInstant());
```

Comment:
Jelly localization requires a `Date` object rather than the (more appropriate) `LocalDate`. Here we downconvert to support Jelly localization.